### ISSUE TITLE
Option to rewrite YT and Twitter links

### ISSUE THREAD

**Issue by dimqua** [searx/searx#1710]
Searx has a plugin system and it should be possible to rewrite YouTube and Twitter links to use Invidious and Nitter respectively.

**Comment by PrasannaVenkadesh** [comment]
Yes. I have implemented a plugin to rewrite youtube and twitter links with custom domain names that you provide in `settings.yml` but I am wondering whether to submit it as a single plugin or one plugin for each service like youtube, twitter, etc., because some might want just youtube links to be rewritten (or) just twitter links to be rewritten.

Once this is decided, I will issue pull request for the same.

**Comment by dimqua** [comment]
> one plugin for each service

I think this is better.

**Comment by return42** [comment]
I closed https://github.com/asciimoo/searx/pull/1726 for now.  After https://github.com/asciimoo/searx/pull/1938 gets merged, this could be implemented as external plugin. I leave that issue open for any further discussion on that topic ...

**Comment by seniorm0ment** [comment]
Worth noting Bibliogram, which is an Instagram frontend.
Also maybe rewrite Google Maps/Apple Maps into OpenStreetMaps 

Edit: also there's Teddit.net now for Reddit

Privacy Redirect extension is good for redirects, however it does not rewrite.

**Comment by 3nprob** [comment]
Noting for others coming here that #1938 was superseded by the now merged #2074 - so there should be no blockers for such a plugin now from what I can tell.

Looks related to #2609
